launcher/net/HttpMetaCache.cpp

```diff
@@ -218,9 +218,24 @@ void HttpMetaCache::Load()
     if (!index.open(QIODevice::ReadOnly))
         return;
 
-    QJsonDocument json = QJsonDocument::fromJson(index.readAll());
+    QJsonParseError parseError;
+    QJsonDocument json = QJsonDocument::fromJson(index.readAll(), &parseError);
+
+    // Fail if the JSON is invalid.
+    if (parseError.error != QJsonParseError::NoError) {
+        qCritical() << QString("Failed to parse HttpMetaCache file: %1 at offset %2")
+                           .arg(parseError.errorString(), QString::number(parseError.offset))
+                           .toUtf8();
+        return;
+    }
+
+    // Make sure the root is an object.
+    if (!json.isObject()) {
+        qCritical() << "HttpMetaCache root should be an object.";
+        return;
+    }
```

Comment:
maybe just start from scratch if the file is invalid. It doesn't store important data anyway so might as well just ignore it and start with a clean state

Reply:
that is what my code does it just prints the logs and exits the load function similar to how it does when it fails to open it.

Reply:
Do you want to drop the logs?

Reply:
I wasn't aware of the context we are in :sweat_smile:

Code is fine then! :D